### PR TITLE
Feat first person controls globe

### PR DESCRIPTION
**Pull request by gliegard**
## Description
FirstPersonControls adapted to Globe view

### CODE REVIEW

**Comment by zarov** [comment]
I'm okay with this change, even if you can, as you told me, add a simpler check in `translateY`, like `view instanceof GlobeView`. I'll let someone else take a look at that too.

**Comment by gliegard** [comment]
@zarov we don't want to test if view is the GlobeView all the time, whereas the view doesn't change. That's why I've created another object !!

**Comment by gliegard** [comment]
@zarov I don't want to test if view is the GlobeView all every time I press the key, whereas the view is not supposed to change.

**Comment by gchoqueux** [comment]
Could you do this in `FirstPersonControls` to make it transparent to the user?

**Comment by gliegard** [comment]
ok, I did the check in the constructor

File: src/Renderer/ThreeExtended/FirstPersonControls.js [hunk 1]
```diff
@@ -161,6 +165,16 @@ class FirstPersonControls extends THREE.EventDispatcher {
         }
     }
 
+    translateY(dt, sign) {
+        this.camera.position.z += sign * this.options.moveSpeed * dt / 1000;
```
Comment (gchoqueux) [inline src/Renderer/ThreeExtended/FirstPersonControls.js]: Strange name `translateY` to add on `z`.
But it is coherent to keep this name?

File: src/Renderer/ThreeExtended/FirstPersonControls.js [hunk 2]
```diff
@@ -161,6 +165,16 @@ class FirstPersonControls extends THREE.EventDispatcher {
...
+    }
+
+    translateYGlobe(dt, sign) {
+        // compute geodesic normale
+        const normal = this.camera.position.clone().normalize();
```
Comment (gchoqueux) [inline src/Renderer/ThreeExtended/FirstPersonControls.js]: You could declare normal, outside the function and use `copy`, instead of `clone`

**Review by gchoqueux — CHANGES_REQUESTED** [review]

File: src/Renderer/ThreeExtended/FirstPersonControls.js (hunk 1 above)
Comment (gliegard) [reply src/Renderer/ThreeExtended/FirstPersonControls.js]: yes, you're right, so I changed method name to 'translateZ'

File: src/Renderer/ThreeExtended/FirstPersonControls.js (hunk 2 above)
Comment (gliegard) [reply src/Renderer/ThreeExtended/FirstPersonControls.js]: yes, good idea !

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**